### PR TITLE
jextract: unstable-2023-11-27 -> unstable-2024-03-13

### LINKED ISSUE #293102: Update request: jextract 2023-11-27 (JDK 21) → 22-jextract+3-13 (2024/2/21)
- Package name: jextract (jextract-unstable)
- Current version on the unstable channel: 2023-11-27
- Current version on the stable/release channel: n/a

- [x] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

As of today the current release build is `22-jextract+3-13 (2024/2/21)`  (see https://jdk.java.net/jextract/) It would be nice for the nixpkgs version to be based on JDK 22 -- which contains the final non-preview release of Java Foreign Function & Memory API (aka "Panama")

There are significant changes to `jextract` that users will want before building/launching projects that use Panama. See: https://cr.openjdk.org/~mcimadamore/panama/jextract_changes.html

I am wiling to contribute but before we can do this we'll need to build with JDK 22 (see Issue #271971)

@SharzyL @SpriteOvO

### PR DESCRIPTION
**Pull request by jlesquembre**
Closes #293102, as requested there, I'm keeping a jextract version compatible with JDK21

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/je/jextract/package.nix [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+    cp -r ./build/jextract $out/opt/jextract
+
+    runHook postInstall
+  '';
+
+  postFixup = ''
+    makeWrapper "$out/opt/jextract/bin/jextract" "$out/bin/jextract"
+  '';
+
+  meta = with lib; {
```
Comment (drupol) [inline pkgs/by-name/je/jextract/package.nix:90]: Missing `meta.platforms`

File: pkgs/by-name/je/jextract/package.nix [hunk 2]
```diff
@@ -0,0 +1,97 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, emptyDirectory
+, writeText
+, makeWrapper
```
Comment (drupol) [inline pkgs/by-name/je/jextract/package.nix:6]: Use preferred `makeBinaryWrapper`.

File: pkgs/by-name/je/jextract/package.nix (hunk 1 above)
Comment (jlesquembre) [reply pkgs/by-name/je/jextract/package.nix:87]: Done

**Review by drupol — APPROVED** [review]
All good for me!

**Pull request merged**